cmake/asan-testing.cmake.in

```diff
@@ -0,0 +1,38 @@
+set(CTEST_SOURCE_DIRECTORY  "@CMAKE_SOURCE_DIR@")
```

Comment:
The 5 files are very similar:
- `cmake/asan-testing.cmake.in`
- `cmake/msan-testing.cmake.in`
- `cmake/tsan-testing.cmake.in`
- `cmake/ubsan-testing.cmake.in`
- `cmake/testing.cmake.in`

The only difference is the value of `CTEST_MEMORYCHECK_TYPE` and the value passed to `-DPCAPPP_USE_SANITIZER`. 

Maybe we can have just 1 file instead of 5 and get this value from an env variable?

If the value is not passed (which is needed if testing without sanitizers) we can avoid setting `CTEST_MEMORYCHECK_TYPE` and `-DPCAPPP_USE_SANITIZER`.

To make things simpler we can have the same value for both (for example: `ThreadSanitizer` for both `CTEST_MEMORYCHECK_TYPE` and `-DPCAPPP_USE_SANITIZER`.

Let me know what you think

Reply:
Yes agree I will try to factorize this part

Reply:
Done